pkg/wsclient/wsclient.go

```diff
@@ -204,6 +209,15 @@ func (w *wsClient) connect(initial bool) error {
 		if w.closed {
 			return false, i18n.NewError(w.ctx, i18n.MsgWSClosing)
 		}
+
+		retry = !initial || attempt < w.initialRetryAttempts
```

Comment:
Want to highlight since it's not obvious in the diff - the logic around `initialRetryAttempts` appears to have been broken in the original code (so that it would only try once to connect on startup before failing). It's tangential to the bulk of the changes in this PR, but it should be working now (with unit test added to demonstrate).